sql-plugin/src/main/scala/org/apache/spark/sql/hive/rapids/GpuHiveTableScanExec.scala

```diff
@@ -535,4 +536,63 @@ class GpuHiveDelimitedTextPartitionReader(conf: Configuration,
       }
     }
   }
+
+  /**
+   * Override of [[com.nvidia.spark.rapids.GpuTextBasedPartitionReader.castStringToDate()]],
+   * to convert parsed string columns to Dates.
+   * Two key differences from the base implementation, to comply with Hive LazySimpleSerDe
+   * semantics:
+   *   1. The input strings are not trimmed of whitespace.
+   *   2. Invalid date strings do not cause exceptions.
+   */
+  override def castStringToDate(input: ColumnVector, dt: DType): ColumnVector = {
+    val cudfFormat = DateUtils.toStrf(dateFormat, parseString = true)
+    withResource(input.isTimestamp(cudfFormat)) { isDate =>
+      withResource(input.asTimestamp(dt, cudfFormat)) { asDate =>
+        withResource(Scalar.fromNull(dt)) { nullScalar =>
+          isDate.ifElse(asDate, nullScalar)
+        }
+      }
+    }
+  }
+
+  // As per Hive's LazySimpleSerDe documentation, only the following timestamp pattern is supported:
```

Comment:
That is odd because the code shows that the format is configurable.

https://github.com/apache/hive/blob/374564f49b345f214ab8845e84244ac4ea4f0668/serde/src/java/org/apache/hadoop/hive/serde2/lazy/LazySerDeParameters.java#L125-L129

At a minimum I think we need to fall back to the CPU if "timestamp.formats" is set in the params.